geth/common/rpcall.go

```diff
@@ -0,0 +1,128 @@
+package common
+
+import (
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// RPCCall represents a unit of a rpc request which is to be executed.
+type RPCCall struct {
+	ID     int64
+	Method string
+	Params []interface{}
+}
+
+// ParseFromAddress returns the address associated with the RPCCall.
+func (r RPCCall) ParseFromAddress() gethcommon.Address {
+	params, ok := r.Params[0].(map[string]interface{})
+	if !ok {
+		return gethcommon.HexToAddress("0x")
```

Comment:
The code was actually taking from the work @farazdagi did, I initially add assume to leave this untouched but I can refactor the `ParseFromAddress` and `ParseToAddress` to return `(gethcommon.Address, error)` to match what we expect.